File: sol/solution/0001-0100/0040. Combination Sum II/Solution.go
func combinationSum2(candidates []int, target int) [][]int {
	sort.Ints(candidates)
	var result [][]int
	var backtrack func(start int, target int, path []int)
	backtrack = func(start int, target int, path []int) {
		if target == 0 {
			temp := make([]int, len(path))
			copy(temp, path)
			result = append(result, temp)
			return
		}
		for i := start; i < len(candidates); i++ {
			if i > start && candidates[i] == candidates[i-1] {
				continue
			}
			if candidates[i] > target {
				break
			}
			backtrack(i+1, target-candidates[i], append(path, candidates[i]))
		}
	}
	backtrack(0, target, []int{})
	return result
}

File: sol/solution/0001-0100/0040. Combination Sum II/Solution.js
/**
 * @param {number[]} candidates
 * @param {number} target
 * @return {number[][]}
 */
var combinationSum2 = function(candidates, target) {
	const result = [];
	candidates.sort((a, b) => a - b);
	const backtrack = (start, remaining, path) => {
		if (remaining === 0) {
			result.push([...path]);
			return;
		}
		for (let i = start; i < candidates.length; i++) {
			if (i > start && candidates[i] === candidates[i - 1]) continue;
			if (candidates[i] > remaining) break;
			
			path.push(candidates[i]);
			backtrack(i + 1, remaining - candidates[i], path);
			path.pop();
		}
	};
	backtrack(0, target, []);
	return result;
};